mopidy_spotify/playlists.py

```diff
@@ -49,13 +53,94 @@ def _get_playlist(self, uri, as_items=False):
             as_items,
         )
 
+    @staticmethod
+    def _get_user_and_playlist_id_from_uri(uri):
+        user_id = uri.split(':')[-3]
+        playlist_id = uri.split(':')[-1]
+        return user_id, playlist_id
+
+    @staticmethod
+    def partitions(lst, n=_chunk_size):
+        for i in range(0, len(lst), n):
+            yield lst[i:i+n]
+
+    @staticmethod
+    def _span(p, l):  # like haskell's Data.List.span
+        i = next((i for i,v in enumerate(l) if not p(v)), len(l))
+        return l[:i], l[i:]
+
+    def _patch_playlist(self, playlist, operations):
+        delta_f = 0
+        delta_t = 0
+        unwind_f = []
+        unwind_t = []
```

Comment:
I'm a bit annoyed about having two of each here, but for moving tracks around, there are stupid edge cases otherwise :/